### DISCUSSION TITLE
修改WebServer.cs的ProcessRequestAsync函数，设置为允许跨域请求，以便直接使用ConfigManager进行修改

### DISCUSSION THREAD

**Discussion by defidead** [Pik-4/HsMod#141] [Ideas (许愿池)]
1.修改WebServer.cs的ProcessRequestAsync函数在开头增加
`
context.Response.Headers.Add("Access-Control-Allow-Origin", "*");
context.Response.Headers.Add("Access-Control-Allow-Methods", "POST, GET, OPTIONS");
context.Response.Headers.Add("Access-Control-Allow-Headers", "Content-Type");
string rawUrLower = request.RawUrl.ToLower();
`
和
`
if (request.HttpMethod == "OPTIONS")
{
    context.Response.StatusCode = 204; // No Content
    context.Response.OutputStream.Close();
    return;
}`

2.重新编译hsmod
3.修改HsModConfigManager中的index.html中的
```
//var ip = 'http://127.0.0.1:58744'
        var ip = ''
```
` var ip = 'http://127.0.0.1:58744`

**Comment by Pik-4** [comment]
我认为CROS跨域不是必须的。你需要把配置管理相关页面放在正确的位置，hsmod会自动解析。还有，你猜为什么这一条js被注释了。
